server/src/main/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCache.java

```diff
@@ -529,6 +530,12 @@ public void refresh(final Set<SegmentId> segmentsToRefresh, final Set<String> da
     }
   }
 
+  @Override
+  void logSegmentsToRefresh(String dataSource, Set<SegmentId> ids)
+  {
+    log.info("Refreshing segments [%s] for datasource [%s]", Iterables.limit(ids, 10), dataSource);
```

Comment:
This log line should mention that this is a sample of the over all refresh segments. 
Also this would be per data source no ? Should we just log 5 segments here ?

Reply:
Yes, this is per datasource. Sure, I can make it log 5 segment ids.

Reply:
How frequently will this method be called?

Reply:
In the worst case this would be called every minute. Generally there aren't segments to be refreshed in each cycle. 
Also, as I mentioned in the description we expected this set to be ideally empty on the Coordinator.